### PR TITLE
op-chain-ops: better comments

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Add comments to ensure that a known bug is commented. See https://github.com/ethereum-optimism/optimism/pull/5682 for a longer description.

Adds a minimal check to make sure this bug is never encountered in practice.

### CODE REVIEW

File: op-chain-ops/crossdomain/legacy_withdrawal.go [hunk 1]
```diff
@@ -57,7 +57,9 @@ func (w *LegacyWithdrawal) Encode() ([]byte, error) {
 	return out, nil
 }
 
-// Decode will decode a serialized LegacyWithdrawal
+// Decode will decode a serialized LegacyWithdrawal. There is a known inconsistency
```
Comment (mslipper) [inline op-chain-ops/crossdomain/legacy_withdrawal.go:60]: Can we add a panic in the `Encode` method as well to prevent improper usage elsewhere?

**Review by mslipper — APPROVED** [review]

File: op-chain-ops/crossdomain/legacy_withdrawal.go (hunk 1 above)
Comment (tynes) [reply op-chain-ops/crossdomain/legacy_withdrawal.go:64]: I've added this now

**Comment by tynes** [comment]
Will rebase `feat/mainnet` on `develop` when we do the next migration rehearsal

**Comment by tynes** [comment]
@mslipper I reverted the commit that adds the `panic` because it will break the tests in a way that would require a large refactor

**Pull request merged**